sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/options/DataLakeFileFlushOptions.java

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.file.datalake.options;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.storage.file.datalake.DataLakeFileAsyncClient;
+import com.azure.storage.file.datalake.DataLakeFileClient;
+import com.azure.storage.file.datalake.models.LeaseAction;
+import com.azure.storage.file.datalake.models.DataLakeRequestConditions;
+import com.azure.storage.file.datalake.models.PathHttpHeaders;
+
+
+/**
+ * Optional parameters for appending data to a file when calling flush() on {@link DataLakeFileClient} and
+ * {@link DataLakeFileAsyncClient}
+ */
+@Fluent
+public class DataLakeFileFlushOptions {
+
+    private Boolean retainUncommittedData;
+    private Boolean close;
+    private PathHttpHeaders pathHttpHeaders;
+    private DataLakeRequestConditions requestConditions;
+    private LeaseAction leaseAction;
+    private Integer leaseDuration;
+    private String proposedLeaseId;
+
+    /**
+     * If "true", uncommitted data is retained after the flush operation completes; otherwise, the uncommitted data is
+     * deleted after the flush operation. The default is false. Data at offsets less than the specified position are
+     * written to the file when flush succeeds, but this optional parameter allows data after the flush position to be
+     * retained for a future flush operation.
+     *
+     * @return whether to retain uncommitted data.
+     */
+    public Boolean getRetainUncommittedData() {
```

Comment:
nit: boolean type, `isUncommittedDataRetained`, and `setUncommittedDataRetained`

Reply:
It appears we are using `Boolean` in other options bags - https://github.com/Azure/azure-sdk-for-java/blob/cdc9319defe333d0d027f63ae8802f3b3a64159f/sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/options/DataLakeFileAppendOptions.java#L72.  I think we should be internally consistent and leave this a `Boolean`.